pkgs/by-name/op/openbao/package.nix

```diff
@@ -0,0 +1,83 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  buildGoModule,
+  installShellFiles,
+  nixosTests,
+  makeWrapper,
+  gawk,
+  glibc,
+  testers,
+  openbao,
+}:
+
+buildGoModule rec {
+  pname = "openbao";
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "openbao";
+    repo = "openbao";
+    rev = "v${version}";
+    hash = "sha256-2dylOra/XuAQMIeb05Lp2nqLxXhPmSK+4poSaZ0GzJs=";
+  };
+
+  vendorHash = "sha256-vdjoYw0VWNdMYJERFj2RAO63WbDShgeyNili5BFEPUk=";
+
+  proxyVendor = true;
+
+  subPackages = [ "." ];
+
+  nativeBuildInputs = [
+    installShellFiles
+    makeWrapper
+  ];
+
+  tags = [
+    "openbao"
+    "bao"
+  ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/openbao/openbao/version.GitCommit=${src.rev}"
+  ];
+
+  postInstall =
+    ''
+      mv $out/bin/openbao $out/bin/bao
+      echo "complete -C $out/bin/bao bao" > bao.bash
+      installShellCompletion bao.bash
+    ''
+    + lib.optionalString stdenv.isLinux ''
+      wrapProgram $out/bin/bao \
+        --prefix PATH ${
+          lib.makeBinPath [
+            gawk
+            glibc
+          ]
+        }
+    '';
+
+  # TODO: Enable the NixOS tests after adding OpenBao as a NixOS service in an upcoming PR and
+  # adding NixOS tests
+  #
+  # passthru.tests = { inherit (nixosTests) vault vault-postgresql vault-dev vault-agent; };
```

Comment:
One other note, it looks like `vault-postgres` tests with Postgres as a storage backend, which this release does not support. OpenBao 2.1.0 [will support Postgres](https://github.com/openbao/openbao/pull/467), barring unforseen issues.